util/update_code

```diff
@@ -0,0 +1,38 @@
+#! /bin/sh
+
+code_src=${1:-code/src}
+
+[ -d ".git" -a -r "book.toml" -a -r "src/tests/chunk_type_tests.rs" ] \
+  || { echo "Run this from the top level of the sources" >&2; exit 2; }
+
+[ -r "$code_src/chunk_type.rs" ] \
+  || { echo "Can't find sources in '$code_src'" >&2; exit 2; }
+
+if [ $# -eq 0 ]; then
+  for stubs in src/stubs/*_full.rs src/stubs/*_stubs.rs; do
+    base=$(basename "$stubs")
+    base=${base%_full.rs}
+    base=${base%_stubs.rs}
+    out=$code_src/$base.rs
+    echo "Setting $out to $stubs"
+    cp "$stubs" "$out"
+    if [ -r src/tests/${base}_tests.rs ]; then
+      printf "\n%s\n" '#[cfg(test)]' >> "$out"
+    fi
+  done
+fi
+
+for tests in src/tests/*_tests.rs; do
+  base=$code_src/$(basename "$tests")
+  base=${base%_tests.rs}
+  out=$base.rs
+  echo "Updating $out with $tests"
+  sed -i -n \
+    "
+      /^#\[cfg(test)]/ {
+        r $tests
+        q
+      }
+      p
+    " "$out"
+  done
```

Comment:
imma be honest here, I don't bash and I never will. I have no way to maintain this 😆

I can't remember why I moved the `code` directory into this repo, but I'm guessing it was so I could use the compiler to check changes I had to make in the book. I'm really nervous about any kind of automated copying from this directory because it gives too much implementation away to the reader.

It's kind of annoying, but the manual work keeps me feeling comfortable about the level of hints provided.